File: docs/manual/parameteroptimization_pde/main_newton.lua
ug_load_script("ug_util.lua")
local pfile = util.GetParam ("-p"			,	"") --gets argument after -p when calling script
local sfile = util.GetParam ("-s"			,	"")

if (pfile ~= "") then
	ug_load_script(pfile)
	print("LOADED "..pfile)
	pFileLoaded = true
end

-----------------------------------------------------------------
-- define Home-Directories
----------------------------------------------------------------
ug4_home        = ug_get_root_path().."/"
app_home        = ug4_home.."apps/parameteroptimization/"
common_scripts  = app_home.."scripts/"
geom_home       = app_home.."geometry/"

-----------------------------------------------------------------

--Defining the parameters
--[[
Parameters that need to be optimized: 
alpha =  ansteckungsrate 
Kappa =  ausbruchsrate 0<=kappa<=1
theta =   0 <=  Sterberate <= 1 -- https://coronavirus.jhu.edu/data/mortality
qq =  Inkubationszeit (in days) (5-6 from RKI (the median is 5-6 days))(95%der infizierten Symtome entwickelt,lag das 95.percentil der Inkubationszeit bei10-14)
pp=  duration of desease

]]

-- Set inital value, lower bound, upper bound for each parameter
v_alpha=EVar64(EFloat64(0.0924629297966215),EFloat64(0.00000001),EFloat64(10))
v_kappa=EVar64(EFloat64(0.1),EFloat64(0),EFloat64(1))
v_theta=EVar64(EFloat64(0.02),EFloat64(0),EFloat64(1))
v_qq=EVar64(EFloat64(14),EFloat64(5),EFloat64(16))
v_pp=EVar64(EFloat64(7),EFloat64(4),EFloat64(15))

--Add the parameters into the Parameter Manager 
--Note: Only the added parameters will be optimized
manager=EVar64Manager()
manager:add("alpha",v_alpha)
--[[manager:add("kappa",v_kappa)
manager:add("theta",v_theta)
manager:add("qq",v_qq)
manager:add("pp",v_pp)
]]
--Defining storage place for the the estimated paramteers
estimated_parameters=EVar64Manager()

--Running Newton Gauss
stepsize_alpha=1.0 --Hyperparameter of the adaptive stepsize calculation. See AdaDelta for details. The second hyperparamater (squared gradient decay rate) can be set by directly interfacing with the C++ code
print(stepsize_alpha)
RunNewtonGauss_BiogasEval("/home/devanshr/gcsc/ug4/apps/parameteroptimization/",manager,estimated_parameters,stepsize_alpha) 


--[[Generate a lua table with the parameters.
Note: The paramters are converted to double which means they lose
their accumulated error bounds.
--]]
tab={}
for k=0,estimated_parameters:len()-1 do
			
			name=estimated_parameters:get_name(k)
			value=estimated_parameters:get_param(k):get_value_as_double()
			tab[name]=value
end
print("Parameter values converted into lua table:")
print(tab)


print("Execution completed") 

File: docs/manual/parameteroptimization_pde/main_newton.cpp
#include<iostream>
#include "/home/devanshr/gcsc/ug4/plugins/ConstrainedOptimization/core/parameter_estimation.h"
#include "/home/devanshr/gcsc/ug4/plugins/ConstrainedOptimization/core/parameters.h"
#include <string>



int main(){

// Set initial values, lower bounds and upperbounds for each parameter
co::EVar64Manager initial_vars;
co::EVar64 alpha(co::EFloat64(0.115636557),co::EFloat64(0.1),co::EFloat64(0.13));
co::EVar64 kappa(co::EFloat64(0.070439024),co::EFloat64(0.05),co::EFloat64(0.1));
co::EVar64 theta(co::EFloat64(0.148484475),co::EFloat64(0.1),co::EFloat64(0.16));
co::EVar64 qq(co::EFloat64(10.33499473),co::EFloat64(8),co::EFloat64(12));
co::EVar64 pp(co::EFloat64(6.130300401),co::EFloat64(4),co::EFloat64(8));

//co::EVar64 c(co::EFloat64(0.6),co::EFloat64(0.6),co::EFloat64(2));
//co::EVar64 k(co::EFloat64(7.52149380283449),co::EFloat64(-50),co::EFloat64(50));

// add the parameters to the EVar64Manager
initial_vars.add("alpha",alpha);
initial_vars.add("kappa",kappa);
initial_vars.add("theta",theta);
initial_vars.add("qq",qq);
initial_vars.add("pp",pp);


co::NewtonOptions options;
options.set_stepsize_alpha(1);
std::string dir ="/home/devanshr/gcsc/ug4/apps/parameteroptimization/";
co::BiogasEvaluation<co::EFloat64,co::ConfigComputation::Local,co::ConfigOutput::File> evaluator(dir, "subset_target.lua","subset_sim.lua");
co::EVarManager<co::EFloat64> estimated_vars;
co::NewtonOptimizer<decltype(evaluator)> solver(options,evaluator);

solver.run(initial_vars,estimated_vars);

// if the machine has a mail server configuered, then notification on completion of optimiztion can be sent per email 
//system("echo \"Optimization Complete\" | mail -s Server1News rastogi@gcsc.uni-frankfurt.de");
}
